build/package.php

```diff
@@ -256,7 +257,15 @@
 						}
 						$source_dir_files = $package->locations['old_versions'] . $package->get('simple_name') . '/' . $source_dir_files;
 						$destination_dir = $dest_filename_dir . '/install/update/new/' . $destination_dir;
-						$package->run_command('cp ' . $source_dir_files . ' ' . $destination_dir);
+
+						if (isset($data['recursive']) && $data['recursive'])
+						{
+							$package->run_command('cp -Rp ' . $source_dir_files . ' ' . $destination_dir);
```

Comment:
It is used in all the other cps, honestly I don't see why would we want to use preserve anywhere, but i thought that's the convention so I followed that.